src/main/java/core/basesyntax/model/OperationFactory.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.model;
+
+import core.basesyntax.db.Warehouse;
+import core.basesyntax.model.entities.Product;
+import core.basesyntax.model.entities.exception.InvalidOperationException;
+import core.basesyntax.model.impl.BalanceOperation;
+import core.basesyntax.model.impl.PurchaseOperation;
+import core.basesyntax.model.impl.ReturnOperation;
+import core.basesyntax.model.impl.SupplyOperation;
+
+public class OperationFactory<T extends Product> {
+    private final Warehouse<T> warehouse;
+
+    public OperationFactory(Warehouse<T> warehouse) {
+        this.warehouse = warehouse;
+    }
```

Comment:
```suggestion
```
let's inject the operationStrategy map instead when creating the instance of OperationFactory in main() method

Reply:
Strategy is redundant in my opinion, why Factory is a bad idea?

Reply:
you can have the factory, just replace switch-case with the map
switch-case is often considered not to be a good solution